lib/ecto/multi.ex

```diff
@@ -0,0 +1,186 @@
+defmodule Ecto.Multi do
+  alias __MODULE__
+  alias Ecto.Changeset
+
+  defstruct operations: [], names: MapSet.new
+
+  @type run :: (t, Keyword.t -> {:ok | :error, any}) | {module, atom, [any]}
+  @type operation :: {:changeset, Changeset.t} |
+                     {:run, run} |
+                     {:update_all, Ecto.Query.t, Keyword.t} |
+                     {:delete_all, Ecto.Query.t}
```

Comment:
Let's make it `{:update_all, Ecto.Query.t, Keyword.t, Keyword.t}` and `{:delete_all, Ecto.Query.t, Keyword.t}` the last one being options.